Extension/src/Automation/Async/manual-promise.ts

```diff
@@ -0,0 +1,102 @@
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All Rights Reserved.
```

Comment:
Can these be put in Extrension/src/Async? i.e. can the Automation folder be removed?

Reply:
My next PR adds ALOT more files into the `Automation` folder that are not in the same directory

Reply:
Sure, but what does the Async folder have to be a subfolder of Automation? Is it Automation related?

Reply:
It's part of a much larger 'automation' suite.  (see https://github.com/fearthecowboy/vscode-cpptools/tree/compiler-detection/Extension/src/Automation ) 

I can promote them all higher, but that makes a lot more folders at the upper level, which seemed less in line with the code that's already there.

Reply:
No, I'm fine with keeping in that subfolder, but why is the folder called "Automation"?

Reply:
renaming to `Utility`